### PR TITLE
Remove unnecessary test scripts from @bentley/build-tools

### PR DESCRIPTION
**Pull request by wgoehrig**
It's time to drop support for `betools test` and `betools test-tsnode`.  These were unnecessary wrappers around mocha that only served to complicate our npm scripts.  Using mocha directly is simpler and much easier to maintain.

Also removed the "docscoverage" script from `@bentley/build-tools` - it was completely undocumented and unused.

### CODE REVIEW

File: example-code/app/package.json [hunk 1]
```diff
@@ -13,7 +13,7 @@
     "extract": "betools extract --fileExt=ts --extractFrom=./src --recursive --out=../../generated-docs/extract",
     "docs": "npm run extract && npm run extract-assets",
     "lint": "eslint -f visualstudio \"./src/**/*.ts\" 1>&2",
-    "test": "npm run copy:test-backend-assets && npm run copy:backend-assets && betools test --defineWindow --testDir=\"./lib/backend/test\"",
+    "test": "npm run copy:test-backend-assets && npm run copy:backend-assets && mocha --no-config",
```
Comment (calebmshafer) [inline example-code/app/package.json:16]: Does the `no-config` option tell mocha to not look for an actual file and force it to use the one in the `package.json`?

**Review by calebmshafer — APPROVED** [review]

**Review by grigasp — APPROVED** [review]

**Review by pmconne — APPROVED** [review]

**Review by kabentley — APPROVED** [review]

**Review by bsteinbk — APPROVED** [review]

**Comment by calebmshafer** [comment]
@Mergifyio update

File: example-code/app/package.json (hunk 1 above)
Comment (wgoehrig) [reply example-code/app/package.json:16]: yeah, basically

**Comment by wgoehrig** [comment]
@Mergifyio update

**Review by jchick-bentley — APPROVED** [review]

**Pull request merged**